ibc/utils.go

```diff
@@ -139,7 +139,7 @@ func IsSingleHop(rawDenom string) bool {
 
 	// Each hop in the path is represented by a pair of port and channel ids
 	// If the number of components in the path is more than 2, it has hopped multiple chains
-	return len(pathComponents) <= 2
+	return len(pathComponents) < 2
```

Comment:
I think this change is false, since a single hop token has more at least two path components because of the port and channel 🤔

Reply:
coin.Denom contains IBC denomination (e.g aevmos, IBC/hash)
data.Denom contains source chain denomination (e.g transfer/channel-x/aevmos, uosmo)

In this function we are checking the source chain denomination (2nd case).
On osmosis chain coins will look like:
`uosmo` <- less than 2 slashes is native from source chain
`transfer/channel-xx/atom` <- 2 slashes is not native. So its a hop coin.

I need to add the check before this function to only check coins that are not native from evmos.
So it should look like 
`is_IBC(coin.Denom) && is_Single_hop(data.Denom) `

Reply:
that makes sense, thanks for the context :+1: